zone/spells.cpp

```diff
@@ -6258,12 +6258,22 @@ void Mob::SendBuffsToClient(Client *c)
 	}
 }
 
-EQApplicationPacket *Mob::MakeBuffsPacket(bool for_target)
+EQApplicationPacket *Mob::MakeBuffsPacket(bool for_target, bool clear)
 {
 	uint32 count = 0;
+	uint32 buff_count;
+
 	// for self we want all buffs, for target, we want to skip song window buffs
 	// since NPCs and pets don't have a song window, we still see it for them :P
-	uint32 buff_count = for_target ? GetMaxBuffSlots() : GetMaxTotalSlots();
+	if (for_target)
+	{
+		buff_count = (clear) ? 0 : GetMaxBuffSlots();
+	}
+	else
+	{
+		buff_count = GetMaxTotalSlots();
+	}
+
```

Comment:
Formatting, one these are preferred.
```cpp
if {
}
else {
}
```
or
```cpp
if {
} else {
}```

Reply:
Could also use a double ternary if you wanted. Not sure if others would approve though.
```cpp
buff_count = for_target ? (clear ? 0 : GetMaxBuffSlots()) : GetMaxTotalSlots();
```

Reply:
Ok, updated this function.